nautobot/ipam/views.py

```diff
@@ -818,6 +840,14 @@ def post(self, request):
             },
         )
 
+    def assign_ipaddresses_to_interface(self, pks, request):
+        ip_addresses = IPAddress.objects.filter(pk__in=pks)
+        interface_model = Interface if "interface" in request.GET else VMInterface
+        interface_id = request.GET.get("interface") or request.GET.get("vminterface")
+        interface = interface_model.objects.get(id=interface_id)
```

Comment:
Do we need to apply queryset `restrict()` to `interface_model.objects` to prevent a user from assigning an IPAddress to an interface that they don't have edit permissions for? Or is that handled magically?

Reply:
Would simply hiding the 'Assign IPAddress' button in the Device List View Table suffice if the user does not have permission to edit the interface?